OpenSim/Simulation/Test/testManager.cpp

```diff
@@ -136,3 +141,83 @@ void testStationCalcWithManager()
         SimTK_TEST_EQ(a, ao);
     }
 }
+
+void testStateChangesBetweenIntegration()
+{
+    using SimTK::Vec3;
+
+    Model model;
+    model.setName("ball");
+
+    auto ball = new Body("ball", 0.7, Vec3(0.1),
+        SimTK::Inertia::sphere(0.5));
+    model.addBody(ball);
+
+    auto freeJoint = new FreeJoint("freeJoint", model.getGround(), Vec3(0), Vec3(0),
+        *ball, Vec3(0), Vec3(0));
+    model.addJoint(freeJoint);
+
+    double g = 9.81;
+    model.setGravity(Vec3(0, -g, 0));
+
+    Station* myStation = new Station();
+    const SimTK::Vec3 point(0);
+    myStation->set_location(point);
+    myStation->setParentFrame(*ball);
+    model.addModelComponent(myStation);
+
+    SimTK::State state = model.initSystem();
+
+    //auto sliderCoord = 
+    //    freeJoint->getCoordinate(FreeJoint::Coord::TranslationY);
+    const Coordinate& sliderCoord = 
+        freeJoint->getCoordinate(FreeJoint::Coord::TranslationY);
+
+    std::vector<double> integInitTimes = {0.0, 1.0, 3.0};
+    std::vector<double> integFinalTimes = {1.0, 3.0, 6.0};
+    std::vector<double> initHeights = {0.0, 13.3, 6.5};
+    std::vector<double> initSpeeds = {0.0, 0.5, -0.5};
+    state.setTime(integInitTimes[0]);
+    int n = integFinalTimes.size();
+
+    for (int i = 0; i < n; ++i) {
+        // Set initial state for integration and check that it's correct
+        sliderCoord.setValue(state, initHeights[i]);
+        sliderCoord.setSpeedValue(state, initSpeeds[i]);
```

Comment:
Changing the state without an explicit realize to at least velocity doesn't seem right. I would always realize after changing the state.  Does that make a difference?

Reply:
I just tried it and the test passes locally.

Reply:
Shouldn't you then get an error that the state is not realized to the appropriate stage?

Reply:
This seems fragile. Also, it seems doubly weird since within the `integrate()` function itself, it just always calls `realizeVelocity()`. See [this line](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Manager/Manager.cpp#L679).

Reply:
> This seems fragile.

I agree.  I also don't know why we don't get an error/exception otherwise. There might be something up with State copying. I wouldn't expect the clone to copy the cache. I agree something is still amiss with the Manager.